client/dashboard/domains/domain-overview/featured-card-site.tsx

```diff
@@ -16,13 +17,35 @@ export default function FeaturedCardSite( { domain }: Props ) {
 		return null;
 	}
 
+	const shouldShowAddAttachSite =
+		domain.is_domain_only_site &&
+		! domain.is_gravatar_restricted_domain &&
+		domain.subtype.id !== DomainSubtype.DOMAIN_TRANSFER;
+
 	return (
 		<OverviewCard
 			title={ __( 'Site' ) }
 			heading={ <span style={ { wordBreak: 'break-all' } }>{ site.name }</span> }
-			link={ `/sites/${ site.slug }` }
+			link={
+				shouldShowAddAttachSite
+					? `/domains/${ domain.domain }/transfer/other-site`
+					: `/sites/${ site.slug }`
+			}
 			icon={ <SiteIcon site={ site } /> }
-			description={ domain.domain }
+			description={ shouldShowAddAttachSite ? __( 'Attach to an existing site' ) : domain.domain }
+			bottom={
+				shouldShowAddAttachSite && (
+					<HStack>
```

Comment:
It would be better to use the `ButtonStack` component for consistency. However, this might be a special case, so it may be fine as is 🤔